bigbluebutton-html5/imports/ui/components/join-handler/presenceManager/utils.tsx

```diff
@@ -0,0 +1,22 @@
+import { useRef } from 'react';
+import { MeetingStatusData } from 'bigbluebutton-html-plugin-sdk';
+import { UpdatedEventDetails } from 'bigbluebutton-html-plugin-sdk/dist/cjs/core/types';
+import { GENERIC_HOOK_PLUGIN, HookEvents } from 'bigbluebutton-html-plugin-sdk/dist/cjs/core/enum';
+
+const updateMeetingStatusForPlugin = (currentUserCurrentlyInMeeting: boolean) => {
+  const previousUserCurrentlyInMeeting = useRef(true);
```

Comment:
Please, either use another approach to store the previous value (for example, a regular top-level variable) since `updateMeetingStatusForPlugin` is neither a hook nor a component, or turn it into a hook.

Reply:
I'm a against the use of top-level variables, that kind of appproach doesn't follow react patterns and make it difficult the extend in futures changes, we have a hook for this implemented on bbb, called [usePreviousValue](https://github.com/Tainan404/bigbluebutton/blob/c9fb031194f1bc9d4d686fcf1ecc734310357c2e/bigbluebutton-html5/imports/ui/hooks/usePreviousValue.ts#L9)